sympy/vector/functions.py

```diff
@@ -214,6 +214,38 @@ def gradient(scalar, coord_sys):
 
     return coord_sys.delop(scalar).doit()
 
+def directional_derivative(scalar, vect, coord_sys):
+    """
+    Returns the directional derivative of a scalar field computed along a given vector
+    in given coordinate system.
+
+    Parameters
+    ==========
+
+    scalar : SymPy Expr
+        The scalar field to compute the gradient of
+
+    vect : Vector
+        The vector operand
+
+    coord_sys : CoordSysCartesian
+        The coordinate system to calculate the gradient in
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, directional_derivative
+    >>> R = CoordSysCartesian('R')
+    >>> f1 = R.x*R.y*R.z
+    >>> v1 = 3*R.i + 4*R.j + R.k
+    >>> directional_derivative(f1, v1, R)
+    R.x*R.y + 4*R.x*R.z + 3*R.y*R.z
+    >>> f2 = 5*R.x**2*R.z
+    >>> directional_derivative(f2, v1, R)
+    5*R.x**2 + 30*R.x*R.z
+
+    """
+    return gradient(scalar,coord_sys).dot(vect).doit()
```

Comment:
@Upabjojr , in reply to your last comment. I would like to point out that in whenever we call the `gradient` function from vector/functions.py then it calls the class `CoordSysCartesian` in vector/coordsysrect.py int his class `delop` function is called. which further calls `Del` function in deloperator.py . Now, the class `CoordSysCartesian`  requires the name of Coordinate system for assigning purposes so that's we are passing coordinate system in gradient function.